### ISSUE TITLE
`pairwise_dxy` problems when given `start` and/or `stop` positions

### ISSUE THREAD

**Issue by travc** [cggh/scikit-allel#75]
`pairwise_dxy` has a problem when given `start` and/or `stop` position.
The unhandled case is actually in `sequence_divergence`.  The slice based on those positions is applied to `ac1` and `ac2`, but not `an1` and `an2`.

Here's a fix (last 4 lines are new)... apologies for not doing a proper pull request.  This is in the `stats/diversity.py` file.

```
    def sequence_divergence(pos, ac1, ac2, an1=None, an2=None, start=None,
                                              stop=None, is_accessible=None):
        # check inputs
        if not isinstance(pos, allel.model.ndarray.SortedIndex):
            pos = allel.model.ndarray.SortedIndex(pos, copy=False)
        if start is not None or stop is not None:
            loc = pos.locate_range(start, stop)
            pos = pos[loc]
            ac1 = ac1[loc]
            ac2 = ac2[loc]
            if an1 is not None:
                an1 = an1[loc]
            if an2 is not None:
                an2 = an2[loc]
...
```

**Comment by alimanfoo** [comment]
Thanks Travis, don't worry about a PR, I'll edit in the fix for the next
release.

On Fri, Mar 11, 2016 at 7:48 PM, Travis Collier <notifications@github.com>
wrote:

> pairwise_dxy has a problem when given start and/or stop position.
> The unhandled case is actually in sequence_divergence. The slice based on
> those positions is applied to ac1 and ac2, but not an1 and an2.
>
> Here's a fix (last 4 lines are new)... apologies for not doing a proper
> pull request. This is in the stats/diversity.py file.
>
>     def sequence_divergence(pos, ac1, ac2, an1=None, an2=None, start=None,
>                                               stop=None, is_accessible=None):
>         # check inputs
>         if not isinstance(pos, allel.model.ndarray.SortedIndex):
>             pos = allel.model.ndarray.SortedIndex(pos, copy=False)
>         if start is not None or stop is not None:
>             loc = pos.locate_range(start, stop)
>             pos = pos[loc]
>             ac1 = ac1[loc]
>             ac2 = ac2[loc]
>             if an1 is not None:
>                 an1 = an1[loc]
>             if an2 is not None:
>                 an2 = an2[loc]
> ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cggh/scikit-allel/issues/75>.
>



-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com <alimanfoo@gmail.com>
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721

**Comment by alimanfoo** [comment]
Resolved in #96.
